src/sfError.ts

```diff
@@ -83,26 +101,39 @@ export class SfError<T = unknown> extends NamedError {
     this.#code = code;
   }
 
+  /** like the constructor, but takes an typed object and let you also set context and data properties */
+  public static create<T extends ErrorDataProperties = ErrorDataProperties>(inputs: SfErrorOptions<T>): SfError<T> {
+    const error = new SfError<T>(inputs.message, inputs.name, inputs.actions, inputs.exitCode, inputs.cause);
+    error.data = inputs.data;
+    error.context = inputs.context;
+    return error;
+  }
   /**
    * Convert an Error to an SfError.
    *
    * @param err The error to convert.
    */
-  public static wrap(err: Error | string): SfError {
+  public static wrap<T extends ErrorDataProperties = ErrorDataProperties>(err: unknown): SfError<T> {
     if (isString(err)) {
-      return new SfError(err);
+      return new SfError<T>(err);
     }
 
     if (err instanceof SfError) {
-      return err;
+      return err as SfError<T>;
     }
 
-    const sfError = new SfError(err.message, err.name, undefined, err);
+    const sfError =
+      err instanceof Error
+        ? // a basic error with message and name.  We make it the cause to preserve any other properties
+          new SfError<T>(err.message, err.name, undefined, err)
+        : // ok, something was throws that wasn't error or string.  Convert it to an Error that preserves the information as the cause and wrap that.
+          SfError.wrap<T>(new TypeError('An unexpected error occurred', { cause: err }));
```

Comment:
Why `TypeError` and not just `Error`?

Reply:
see above comment about TypeError.  Sounds like it's confusing to 2 people now.

I'd argue it's fairly idiomatic in TS (typically, type-checking an any/unknown)
see https://github.com/search?q=throw+new+TypeError+language%3ATypeScript&type=code&l=TypeScript

Reply:
To me, a TypeError means that the author of the code made a mistake and someone needs to fix a client side bug.  Usually this would mean the CLI team.  W.r.t "plugin health" it would be categorized as a bug for the CLI team (if called from the sf CLI).  If this is intended, then it's fine.  If the meaning of the error is anything else, we should just use `Error`.  Although, since it's immediately being wrapped by SfError the command framework probably wouldn't interpret this as a TypeError and would categorize it differently.  In the end it probably doesn't matter much.